### PR TITLE
Globally deny Clippy and `rust_2018_idioms`

### PR DESCRIPTION
**Pull request by ojeda**

### CODE REVIEW

**Comment by ksquirrel** [comment]
Review of `fd5fe68a9c8b`:
  - ⚠️ The PR body message looks too short.
  - ✔️ Commit [`ca7d3dc`](https://github.com/Rust-for-Linux/linux/commit/ca7d3dcb4d8241099f36253d14eac56e968bcac3): Looks fine!
  - ✔️ Commit [`01d62e8`](https://github.com/Rust-for-Linux/linux/commit/01d62e8b17ddd176e85cebcafb3b75094a226894): Looks fine!
  - ✔️ Commit [`fd5fe68`](https://github.com/Rust-for-Linux/linux/commit/fd5fe68a9c8bc016521af5fba4c76ad21e5da07b): Looks fine!

**Comment by ojeda** [comment]
@wedsonaf This includes a small change to be Clippy warning free now that it applies globally, i.e. to kernel modules too.

Also, there were a few cleanups required for `rust_2018_idioms` in Binder, so I decided to go with an `allow` in Binder for the moment -- it would be great to clean them up whenever possible, but there is no rush.

**Comment by ojeda** [comment]
The Clippy part was triggered thanks to @flip1995 that made me notice that https://github.com/rust-lang/rust-clippy/issues/6799 was working.

File: drivers/android/rust_binder.rs [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 #![no_std]
 #![feature(global_asm, try_reserve, allocator_api, concat_idents)]
+#![allow(rust_2018_idioms)]
```
Comment (nbdd0121) [inline drivers/android/rust_binder.rs:10]: Maybe add a FIXME or TODO here?

**Comment by ksquirrel** [comment]
Review of `2e3735fb0d6d`:
  - ⚠️ The PR body message looks too short.
  - ✔️ Commit [`b94a010`](https://github.com/Rust-for-Linux/linux/commit/b94a010cbf2e8e8e1e176b37095540895db70bf4): Looks fine!
  - ✔️ Commit [`349b967`](https://github.com/Rust-for-Linux/linux/commit/349b967bca8ff89729ba5c6d111edf04978e497b): Looks fine!
  - ✔️ Commit [`9b15640`](https://github.com/Rust-for-Linux/linux/commit/9b156409f729e10a3cbd5da6a0aee600e3ce040d): Looks fine!
  - ✔️ Commit [`2e3735f`](https://github.com/Rust-for-Linux/linux/commit/2e3735fb0d6dbc07686120f5d1192e8005e1a105): Looks fine!

File: drivers/android/rust_binder.rs (hunk 1 above)
Comment (ojeda) [reply drivers/android/rust_binder.rs:10]: Added.

**Comment by ksquirrel** [comment]
Review of `64fadd50d631`:
  - ⚠️ The PR body message looks too short.
  - ✔️ Commit [`b94a010`](https://github.com/Rust-for-Linux/linux/commit/b94a010cbf2e8e8e1e176b37095540895db70bf4): Looks fine!
  - ✔️ Commit [`349b967`](https://github.com/Rust-for-Linux/linux/commit/349b967bca8ff89729ba5c6d111edf04978e497b): Looks fine!
  - ✔️ Commit [`9b15640`](https://github.com/Rust-for-Linux/linux/commit/9b156409f729e10a3cbd5da6a0aee600e3ce040d): Looks fine!
  - ✔️ Commit [`64fadd5`](https://github.com/Rust-for-Linux/linux/commit/64fadd50d6311abb620fbdc8cf9183cba5ee5ada): Looks fine!

**Comment by ksquirrel** [comment]
Review of `b548e03e49f2`:
  - ⚠️ The PR body message looks too short.
  - ✔️ Commit [`b94a010`](https://github.com/Rust-for-Linux/linux/commit/b94a010cbf2e8e8e1e176b37095540895db70bf4): Looks fine!
  - ✔️ Commit [`349b967`](https://github.com/Rust-for-Linux/linux/commit/349b967bca8ff89729ba5c6d111edf04978e497b): Looks fine!
  - ✔️ Commit [`9b15640`](https://github.com/Rust-for-Linux/linux/commit/9b156409f729e10a3cbd5da6a0aee600e3ce040d): Looks fine!
  - ✔️ Commit [`b548e03`](https://github.com/Rust-for-Linux/linux/commit/b548e03e49f27a56453b0077c369098174630575): Looks fine!

**Pull request merged**